src/concern.rs

```diff
@@ -152,6 +152,10 @@ impl Acknowledgment {
 }
 
 impl WriteConcern {
+    pub(crate) fn is_acknowledged(&self) -> bool {
+        self.w != Some(Acknowledgment::Nodes(0))
```

Comment:
Per [the spec](https://github.com/mongodb/specifications/blob/master/source/read-write-concern/read-write-concern.rst#unacknowledged-writeconcern), I think this also needs to check that `journal` is not true.

Reply:
Originally I didn't bother because `w: 0, j: true` was invalid, but outside of uri parsing, we were'nt actually validating any write concerns. To fix this, I went back to operations exposing the actual write concern and had `execute_operation` validate it. I also made `is_acknowledged` verify `j: false` for completeness.

Also, instead of maintaining a separate `is_acknowledged` for write concerns specified in `RunCommand`, I just added a `Deserialize` implementation to `WriteConcern` and leveraged that.